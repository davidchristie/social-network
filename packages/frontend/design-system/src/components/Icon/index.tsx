import MaterialIcon from "@material-ui/core/Icon";
import "material-design-icons/iconfont/material-icons.css";
import React from "react";

interface Props {
  children: string;
  size?: "inherit" | "default" | "small" | "large";
  variant?: "action" | "disabled" | "error" | "inherit" | "primary" | "secondary";
}

const Icon: React.SFC<Props> = ({
  size = "small",
  variant = "inherit",
  ...attributes
}) => {
  return (
    <MaterialIcon color={variant} fontSize={size} {...attributes} />
  );
};

export default Icon;
