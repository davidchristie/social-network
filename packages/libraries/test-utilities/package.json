{
  "name": "test-utilities",
  "private": true,
  "scripts": {
    "build": "rimraf dist && tsc"
  },
  "version": "1.0.0"
}
