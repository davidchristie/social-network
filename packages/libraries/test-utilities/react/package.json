{
  "main": "../dist/react/index.js",
  "name": "test-utilities/react",
  "private": true,
  "version": "1.0.0"
}
