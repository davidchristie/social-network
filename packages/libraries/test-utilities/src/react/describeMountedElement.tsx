import { mount } from "enzyme";
import React from "react";

export default function describeMountedElement (
  name: string,
  element: React.ReactElement<any>
) {
  describe(name, () => {
    it("renders without errors", () => {
      mount(element);
    });

    it("matches shapshot", () => {
      const wrapper = mount(element);
      expect(wrapper).toMatchSnapshot();
    });
  });
}
