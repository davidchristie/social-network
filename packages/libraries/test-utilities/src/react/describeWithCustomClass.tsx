import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

interface Props {
  className?: string;
}

export default function describeWithCustomClass (
  Component: React.ComponentType<Props>
) {
  describe("with custom class", () => {
    const CUSTOM_CLASS = "CUSTOM_CLASS";

    let wrapper: ShallowWrapper;

    beforeEach(() => {
      wrapper = shallow(
        <Component className={CUSTOM_CLASS} />,
      );
    });

    it("has custom class", () => {
      expect(wrapper.hasClass(CUSTOM_CLASS)).toBe(true);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
}
