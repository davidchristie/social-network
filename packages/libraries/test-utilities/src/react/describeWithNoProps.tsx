import React from "react";

import describeWithProps from "./describeWithProps";

export default function describeWithNoProps (
  Component: React.ComponentType
) {
  describeWithProps("with no props", Component, {});
}
