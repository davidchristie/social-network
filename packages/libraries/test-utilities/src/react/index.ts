export { default as describeMountedElement } from "./describeMountedElement";
export { default as describeWithCustomClass } from "./describeWithCustomClass";
export { default as describeWithNoProps } from "./describeWithNoProps";
export { default as describeWithProps } from "./describeWithProps";
export { default as itMatchesSnapshot } from "./itMatchesSnapshot";
