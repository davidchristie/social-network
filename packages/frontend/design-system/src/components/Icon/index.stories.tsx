import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Icon from ".";
import Theme from "../Theme";

const stories = storiesOf("Components", module);
stories.addDecorator(getStory => <Theme>{getStory()}</Theme>);
stories.addDecorator(withInfo({
  inline: true,
  propTablesExclude: [Theme],
}));
stories.add(
  "Icon",
  () => <Icon size="large" variant="primary">account_circle</Icon>
);
