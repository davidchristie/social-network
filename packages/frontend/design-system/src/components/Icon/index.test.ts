import { describeWithProps } from "test-utilities/react";

import Icon from ".";

describe("Icon component", () => {
  describeWithProps("with default props", Icon, {
    children: "account_circle",
  });

  describeWithProps(`with variant "primary"`, Icon, {
    children: "account_circle",
    variant: "primary",
  });

  describeWithProps(`with size "large"`, Icon, {
    children: "account_circle",
    size: "large",
  });
});
