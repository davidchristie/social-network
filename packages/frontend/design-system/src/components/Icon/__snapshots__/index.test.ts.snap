// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component with default props matches snapshot 1`] = `
<WithStyles(Icon)
  color="inherit"
  fontSize="small"
>
  account_circle
</WithStyles(Icon)>
`;

exports[`Icon component with size "large" matches snapshot 1`] = `
<WithStyles(Icon)
  color="inherit"
  fontSize="large"
>
  account_circle
</WithStyles(Icon)>
`;

exports[`Icon component with variant "primary" matches snapshot 1`] = `
<WithStyles(Icon)
  color="primary"
  fontSize="small"
>
  account_circle
</WithStyles(Icon)>
`;
